flink-python/pyflink/fn_execution/beam/beam_worker_pool_service.py

```diff
@@ -155,7 +159,12 @@ def _start_sdk_worker_main(self, start_worker_request: beam_fn_api_pb2.StartWork
             _LOGGER.exception('Python sdk harness failed: ')
             raise
         finally:
-            if base_dir:
-                os.chdir(base_dir)
+            self._parse_param_lock.acquire()
+            self._ref_cnt -= 1
+            # The last thread to exit is responsible for changing working directory.
+            if self._ref_cnt == 0 and self._old_working_dir is not None:
```

Comment:
What's the motivation for this? I'm asking because I'm wondering that if we need to change back the worker directory, should we also remove the changed PYTHONPATH?